src/Cardano/AddressV3.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "AddressV3.h"
+#include "AddressV2.h"
+#include "../Data.h"
+#include "../Bech32.h"
+#include "../Base32.h"
+#include "../Crc.h"
+#include "../HexCoding.h"
+#include "../Hash.h"
+
+#include <array>
+
+using namespace TW;
+using namespace TW::Cardano;
+using namespace std;
+
+bool AddressV3::parseAndCheckV3(const std::string& addr, Discrimination& discrimination, Kind& kind, Data& key1, Data& key2) {
```

Comment:
maybe we can return tuple? too many inout parameters

Reply:
Suggestion worth considering, after checking usage I leave it like this though.